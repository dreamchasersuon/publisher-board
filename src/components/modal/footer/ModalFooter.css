.Modal-footer {
  display: flex;
  align-items: center;
  justify-content: center;
  border-top: 1px solid #e5e5e5;
}
