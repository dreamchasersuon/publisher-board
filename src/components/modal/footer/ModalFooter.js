import React from 'react';
import PropTypes from 'prop-types';
import Spinner from '../../spinner/Spinner';
import Button from '../../buttons/button-modal-action/Button';
import './ModalFooter.css';

export default function ModalFooter({
  btnCloseLabel,
  btnProceedLabel,
  handleOnCloseModal,
  handleOnSubmitForm,
  isLoading,
}) {
  return (
    <footer className="Modal-footer">
      {btnCloseLabel && (
        <Button onPress={handleOnCloseModal} label={btnCloseLabel} />
      )}
      {btnProceedLabel && (
        <Button
          testId="modal-perform"
          onPress={handleOnSubmitForm}
          label={isLoading ? <Spinner /> : btnProceedLabel}
        />
      )}
    </footer>
  );
}

ModalFooter.propTypes = {
  btnCloseLabel: PropTypes.string,
  btnProceedLabel: PropTypes.string,
  handleOnCloseModal: PropTypes.func,
  handleOnSubmitForm: PropTypes.func,
  isLoading: PropTypes.bool,
};
