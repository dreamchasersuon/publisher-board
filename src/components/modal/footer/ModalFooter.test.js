import React from 'react';
import snapshotDiff from 'snapshot-diff';
import ModalFooter from './ModalFooter';

test('renders modal footer with different buttons', () => {
  expect(
    snapshotDiff(
      <ModalFooter btnCloseLabel="close" />,
      <ModalFooter btnCloseLabel="close" btnProceedLabel="update" />,
    ),
  ).toMatchSnapshot();
});
