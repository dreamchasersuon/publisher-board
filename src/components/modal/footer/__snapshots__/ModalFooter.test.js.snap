// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders modal footer with different buttons 1`] = `
"Snapshot Diff:
- <ModalFooter btnCloseLabel=\\"close\\" />
+ <ModalFooter btnCloseLabel=\\"close\\" btnProceedLabel=\\"update\\" />

@@ -5,6 +5,12 @@
      className=\\"Modal-action\\"
      type=\\"button\\"
    >
      close
    </button>
+   <button
+     className=\\"Modal-action\\"
+     type=\\"button\\"
+   >
+     update
+   </button>
  </footer>"
`;
